### ISSUE TITLE
Optic doesn't support Symbol.

### ISSUE THREAD

**Issue by jituanlin** [calmm-js/partial.lenses#166]
Well, Optic is support String, Array ...etc 
However, the ES6 new type Symbol is mostly support by modern browser and Node.js.
But, if we create a new Optic like this:
```js
[
  Symbol(),
  'string'
]
```
The error is throw.

**Comment by polytypic** [comment]
Yes, currently Partial Lenses do not support symbols and in non-production mode the library raises an exception as seen [here](https://calmm-js.github.io/partial.lenses/playground.html#DIOg5gpgLgFAygTwLYCMD2AbGBKANAAgG8BfbIA).

I'm not opposed to supporting symbols, but I'm not entirely sure what the best way to support them would be.  [Ramda, for example, does not currently support symbols](https://github.com/ramda/ramda/issues/2137).  It seems that [the use case of symbols is to avoid name clashes](https://stackoverflow.com/questions/21724326/what-is-the-motivation-for-bringing-symbols-to-es6).  Even JavaScript itself seems rather undecided on how symbol properties should be treated: object spread ignores them, while `Object.assign` copies them as seen [here](https://calmm-js.github.io/partial.lenses/playground.html#MYewdgzgLgBAZiEMC8MDKBPAtgIxAGwAoByBEYgSgChRJYAPFGAbwG0yBdALhgEYAGXgF8q+EAHNCLAHSyyAGhgYewmNTGSYAeRwArAKbAo0gIYQIAS3FhCzIYvqLmyvkIpqgA).

I did some quick tests and it seems that copying symbol properties (using builtin `Object.assign` or `Object.getOwnPropertySymbols`) would cause something like a 2x performance hit in latest Node for property lenses.  Note that I mean this in the sense of supporting the possibility of having symbol properties&mdash;I actually tested this on objects that did not have symbol properties, so there was no extra cost of actually copying more.  Hopefully future JS engines will have faster implementations of `Object.assign` and then supporting symbol properties could be done without taking a performance hit.

For the above tow reasons, namely

* that there currently does not seem to be a consensus on what the best way to support them is (copy or not copy), and
* that copying them would cause a 2x performance hit,

I'd rather not enhance the default property lenses in Partial Lenses to support symbol properties at this point.

Note that you can always write a custom optics for accessing objects with symbol properties.  Here is [a simple example using `L.lens`](https://calmm-js.github.io/partial.lenses/playground.html#MYewdgzgLgBAZiEMC8MDKBPAtgIxAGwAoByBEYgSgCgrRJYJsUYIUA+GAGQDp8BTSISoAPdjDABXfPhgBCVKID8MYQG0IAXRgAuGBLAATPnACWYPgYA0VQgDdLKimIDeI5gHkcAKz7Ao3AEMICBMAczBCZwBfB2FqEzgYQn0jU3MDFGRUW2ojfig+FXUNKj58CD4RYuZbKgAnPigJOrAVKiiqaip8EFDCLm5QxsJGLEIyCgdnVTINXQBGAAZ5qKcunr6BhqwQWz4R7HHESZhp2YXl1ZhqDf6eCqgDsYmHABYAJimZxDmYJZW1t1enduA8niRhJQ3p9Tt8QL9-lcKEA) and [here is a version that also provides the symbol or string as the index](https://calmm-js.github.io/partial.lenses/playground.html#MYewdgzgLgBAZiEMC8MDKBPAtgIxAGwAoByBEYgSgCgrRJYJsUYIUA+GQgDwBoYB9AJZ8AYny6CATBhEV2MEQDosAQwAOhKgDd5AbypdmAeRwArAKbAoilRAiCA5mEK6AvuOqC4nAK5gAJuZwgmDm-ijIqFrUgfjmUOYwXADaEAC6VOb4EOYGqWnMWlQATvE+xWBJVO4GUjKEEHxgPvj4MACEqIYA-En5MABcMH6BwaH+1NRU+CAOhDAAMooO8Q3YhGQUfLrJZGlDAIwADAeuclMzc4uKpVggWuZrWBuIWzA7e4cnZzDUl-NLHJQJ4vEBvAAskm2u0Q+xgx1O52mswBiiBIOIXEofEh0M+8O+5yAA).  At the moment this is what I'd recommend if you need to access objects with symbol properties.

**Comment by jituanlin** [comment]
I read your example. However, the variable name like xi2yF confuse me. What it's name mean? When I want to read this project source code. This problem tangle me, I want to contribute my code, but first , how to understand this, which knowledge I miss. Thanks

**Comment by polytypic** [comment]
Also, my latter example had a bug&mdash;I just fixed the link.

**Comment by jituanlin** [comment]
Got it ,thanks

**Comment by polytypic** [comment]
@Justin-ZS Oops.  Thanks for the correction!  Here is the [fixed playground](https://calmm-js.github.io/partial.lenses/playground.html#MYewdgzgLgBAZiEMC8MDKBPAtgIxAGwAoByBEYgSgChRJYAPFGAbwG0yBdALhgEYAGXgF8q+EAHNCLAHSz6AGhgYewmNTGSYAeRwArAKbAo0gIYQIAS3FhCzIYoUtlfIRTVA).

**Comment by ftaiolivista** [comment]
If someone, like me, needs to not loose Symbols using lenses this a patch I have done to infestines.
```

diff --git a/node_modules/infestines/dist/infestines.cjs.js b/node_modules/infestines/dist/infestines.cjs.js
index 3d711b4..a1a4f30 100644
--- a/node_modules/infestines/dist/infestines.cjs.js
+++ b/node_modules/infestines/dist/infestines.cjs.js
@@ -438,14 +438,8 @@ var assocPartialU = function assocPartial(k, v, o) {
   var r = {};
   if (o instanceof Object) {
     if (!isObject(o)) o = toObject(o);
-    for (var l in o) {
-      if (l !== k) {
-        r[l] = o[l];
-      } else {
-        r[k] = v;
-        k = void 0;
-      }
-    }
+    // Symbol Support
+    Object.assign(r, o);
   }
   if (isDefined(k)) r[k] = v;
   return r;
@@ -456,10 +450,9 @@ var dissocPartialU = function dissocPartial(k, o) {
   if (o instanceof Object) {
     if (!isObject(o)) o = toObject(o);
     for (var l in o) {
-      if (l !== k) {
-        if (!r) r = {};
-        r[l] = o[l];
-      } else {
+      // Symbol Support
+      if (!r) {r = {...o}}
+      if (l === k) {
         k = void 0;
       }
     }

```
